src/lints/trait_default_impl_removed.ron

```diff
@@ -0,0 +1,62 @@
+SemverQuery(
+    id: "trait_default_impl_removed",
+    human_readable_name: "pub trait default method impl removed",
+    description: "A non-sealed public trait default method impl was removed",
+    required_update: Major,
+    lint_level: Deny,
+    reference_link: Some("https://doc.rust-lang.org/book/ch10-02-traits.html#default-implementations"),
+    query: r#"
+    {
+        CrateDiff {
+            baseline {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"]) @output
+                        sealed @filter(op: "!=", value: ["$true"])
+                        name @output
+
+                        importable_path {
+                            path @output @tag
+                            public_api @filter(op: "=", value: ["$true"])
+                        }
+
+                        method {
+                            method_name: name @output @tag
+                            public_api_eligible @filter(op: "=", value: ["$true"])
+                            has_body @filter(op: "=", value: ["$true"])
+                        }
+                    }
+                }
+            }
+            current {
+                item {
+                    ... on Trait {
+                        visibility_limit @filter(op: "=", value: ["$public"])
+                        sealed @filter(op: "!=", value: ["$true"])
```

Comment:
This clause poses an interesting question: if the trait newly became sealed _and_ a default impl was removed, is that two breaking changes or one? Should it trigger two lints, or just one?

My gut feeling is that it should trigger two lints, which would mean we don't actually want this clause: regardless of whether the _new_ trait is sealed, we want to complain about the missing method default impl.
```suggestion
```

What do you think?

P.S.: This edge case is super important to cover with tests! It's the sort of thing that can really sneak up on you!